client/server/pages/index.js

```diff
@@ -485,6 +485,26 @@ function setUpCSP( req, res, next ) {
 }
 
 function setUpRoute( req, res, next ) {
+	if ( req.context.isRouteSetup === true ) {
+		try {
+			req.logger.warn(
+				{
+					isLoggedIn: req.context.isLoggedIn,
+					path: req.context.path,
+				},
+				'Route already set up. Ambiguous route definition likely.'
+			);
+		} catch ( err ) {
```

Comment:
I don't like the ergonomics of this try/catch, but it's how this method is used elsewhere, so I guess we should keep it.

If we really need the try/catch, we could improve the ergonomics so we don't need to put it everywhere we need to do a simple log in the code

Reply:
I ended up removing this because I found at least one other case where the try/catch wasn't added.

However, I'm still very confused about what would make `req.logger.info` ever throw an error in the first place. And beyond that, why we would need a try/catch in only some cases. And even more beyond that, why error handling isn't handled at a lower level rather than passing it on to the consumer.

Reply:
Looking at Bunyan sources, it seems that `req.logger.info` can throw an error only when the logger is somehow misconfigured.

There is a try/catch around `req.logger.info` in the `client/server/render` module, in the `render` function, but that has a history going before Bunyan logging was even added. Before Bunyan, there was a `store.dispatch( logToLogstash )` call. See the #34718 PR where that logging was originally added.

Then there's another case in `renderServerError` (`client/server/pages`) that was added in #46307, in addition to original `console.log`.

Both try/catch-es don't seem to serve a specific purpose -- they are either legacy, kept in place when logging method got changed, or "just in case".

Reply:
Cool, thanks for verifying!